### DISCUSSION TITLE
Configuration from AWS?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [basecamp/kamal#784] [Q&A]
Config parameter and secrets are **very cheaply** stored in [AWS Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) or even [AWS S3 buckets](https://www.reddit.com/r/aws/comments/k1gjdm/s3_vs_parameter_store_for_storing_secrets/). Both are very popular and affordable. Is this available in Kamal?

**Comment by nickhammond** [comment]
@hrstoyanov You can pull your secrets from anywhere you'd like, Kamal doesn't have anything built in for this though.

You'd follow this part of the generated _.env_ file instructions and then swap out the `op` calls with AWS CLI calls instead.

https://kamal-deploy.org/docs/configuration/environment-variables/#using-a-generated-env-file

Here's an example utilizing AWS Secrets Manager with the `aws-sdk-secretsmanager` gem, you could easily swap this with the parameter store or S3 CLI commands instead though.

```
<%
  require 'aws-sdk-secretsmanager'
  client = Aws::SecretsManager::Client.new(profile: "staging", region: "us-east-2")
-%>

<% client.list_secrets.secret_list.each do |secret| -%>
  <% JSON.parse(client.get_secret_value(secret_id: secret.arn).secret_string).each_pair do |k, v| -%>
    <%= k %>=<%= v %>
  <% end -%>
<% end -%>

**Comment by anatoly-pashin-axon21** [comment]
Solution for v2 (.kamal/secrets file):
```
KAMAL_REGISTRY_USERNAME=$(aws --profile=$AWS_PROFILE ssm get-parameter --name "$SSM_PREFIX""KAMAL_REGISTRY_USERNAME" --with-decryption --output text --query 'Parameter.Value')
```
